### PR TITLE
chore(pkger): add stack state management for variables

### PR DESCRIPTION
**Pull request by jsteenb2**
references: #17434 

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: pkger/service.go [hunk 1]
```diff
@@ -889,32 +889,32 @@ func (s *Service) dryRunVariables(ctx context.Context, orgID influxdb.ID, pkg *P
 	mExistingLabels := make(map[string]DiffVariable)
 	variables := pkg.variables()
 
-VarLoop:
+	existingVars, _ := s.varSVC.FindVariables(ctx, influxdb.VariableFilter{
+		OrganizationID: &orgID,
+		// TODO: would be ideal to extend find variables to allow for a name matcher
```
Comment (jademcgough) [inline pkger/service.go:894]: Is it worth doing this now? Will this not work correctly for users with 100+ variables?

Comment (jsteenb2) [reply pkger/service.go:894]: we only make this call once per run. It could be done now, but definitely a nice to have.

Comment (jsteenb2) [reply pkger/service.go:894]: the VarSVC doesn't support a find by name atm 😢

Comment (jsteenb2) [reply pkger/service.go]: @jademcgough I pushed up the pagination changes to make sure we get all variables instead of only the first 100 👍

**Pull request merged**